### PR TITLE
Add generic msDebugger && Fetch all wflows with missing TotalInput parameters

### LINKED ISSUE #11183: Missing TotalInput[Events, Lumis, Files] params in production workflows
**Impact of the bug**
Workflows (even fulfilling the number of events requested) aren't complete

**Describe the bug**
[task_HIG-RunIISummer20UL17wmLHEGEN-03631](https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=task_HIG-RunIISummer20UL17wmLHEGEN-03631)

[task_HIG-RunIISummer20UL16wmLHEGENAPV-03387](https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=task_HIG-RunIISummer20UL16wmLHEGENAPV-03387)

[task_HIG-RunIISummer20UL16wmLHEGENAPV-03384](https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=task_HIG-RunIISummer20UL16wmLHEGENAPV-03384)
When checking these 3 WFs, the number of events requested is each 8M, 5M and 270K but no TotalInputEvents in their reqMgr2 page 

[task_HIG-RunIISummer20UL16wmLHEGENAPV-05608](https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=task_HIG-RunIISummer20UL16wmLHEGENAPV-05608)
One more example in which the production is complete but not moving on to the next status due to the issue

**How to reproduce it**
You can check the logs [here](https://cms-unified.web.cern.ch/cms-unified/logs/checkor/2022-06-15_13:56:01.log)
and the JIRA tickets for WFs having this issue [here](https://its.cern.ch/jira/issues/?jql=labels%20%3D%20TotalInputEventsMissing)

**Expected behavior**
The parameter should be seen

@todor-ivanov Can you please have a look?

### PR DESCRIPTION
**Pull request by todor-ivanov**
Fixes #11183

#### Status
In devlopment

#### Description
With the current script we can instantiate a generic micro service object. And then use it to start fetching  all kinds of information from central services we need the same way as we do with the rest of the micro services. This is a minimalistic setup that  allows also for interactive use through a python shell.

Further it is used to fetch and parse all workflows with missing `TotalInput*` parameters.    

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
None

#### External dependencies / deployment changes
None

### CODE REVIEW

File: bin/adhoc-scripts/fetchFromReqmgr.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+                wflow['tmp']['TotalInputEvents'] = totalEvents
+                wflow['tmp']['TotalInputLumis'] = totalLumis
+                wflow['tmp']['TotalEstimatedJobs'] = totalEstimatedJobs
+                wflow['tmp']['TotalInputFiles'] = 0
+                logger.info('\n%s', pformat(wflow['tmp']))
+
+            elif wflow['tmp']['SplittingAlgo'] == 'EventAwareLumiBased':
+                logger.info("Found a workflow with EventAwareLumiBased Splitting algorithm:")
+                keySetInputLists = {'LumiList', 'RunList', 'BlockList'}
```
Comment (amaltaro) [inline bin/adhoc-scripts/fetchFromReqmgr.py:200]: These parameters do not exist in a ReqMgr2 spec.

File: bin/adhoc-scripts/fetchFromReqmgr.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+                for key in keySetInputLists:
+                    try:
+                        wflow['tmp'][key] = wflow[key]
+                    except KeyError:
+                        wflow['tmp'][key] = None
+
+                if any([wflow['tmp'][key] for key in keySetInputLists]):
+                    logger.ino("Found input Lists for th e current workflow. Continue")
```
Comment (amaltaro) [inline bin/adhoc-scripts/fetchFromReqmgr.py:208]: bug!

File: bin/adhoc-scripts/fetchFromReqmgr.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+                # Fetch all datasetFile info. The result should be a list of dictionaries (one per file) of the kind:
+                # [...
+                #  {'Lumis': {1: [209, 230, 21, 181, 25, 146, 232, 74, 191, 207]},
+                #   'NumberOfEvents': 9289,
+                #   'ValidFile': 1},
+                #  {'Lumis': {1: [220, 19, 130, 154, 229]},
+                #   'NumberOfEvents': 4645,
+                #   'ValidFile': 1},
+                # ...]
+                datasetFileInfo = [{key: file[key] for key in ['LogicalFileName', 'Lumis', 'NumberOfEvents', 'ValidFile']}
```
Comment (amaltaro) [inline bin/adhoc-scripts/fetchFromReqmgr.py:221]: why don't you simply replace all this logic by:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/DBS/DBS3Reader.py#L405

? Thus, using the DBS `filesummaries` as initially suggested.

File: bin/adhoc-scripts/fetchFromReqmgr.py (hunk 1 above)
Comment (todor-ivanov) [reply bin/adhoc-scripts/fetchFromReqmgr.py:200]: fixed

File: bin/adhoc-scripts/fetchFromReqmgr.py (hunk 2 above)
Comment (todor-ivanov) [reply bin/adhoc-scripts/fetchFromReqmgr.py:208]: thanks, just fixed that

File: bin/adhoc-scripts/fetchFromReqmgr.py (hunk 3 above)
Comment (todor-ivanov) [reply bin/adhoc-scripts/fetchFromReqmgr.py:221]: This is indeed much better.  Thanks @amaltaro . I am switching to this method now.

**Comment by amaltaro** [comment]
test this please

**Review by amaltaro — APPROVED** [review]

**Pull request merged**